sonic_platform_base/sonic_pcie/pcie_common.py

```diff
@@ -77,9 +83,26 @@ def get_pcie_device(self):
                     pciList.append(pciDict)
                     pciDict = deepcopy(pciDict)
                 else:
-                    print("CAN NOT MATCH PCIe DEVICE")
+                    log.log_warning("CAN NOT MATCH PCIe DEVICE")
         return pciList
 
+    # Check device ID from procfs file check for each PCI device
+    def check_pcie_deviceid(self, bus="", device="", fn="", id=""):
+        current_file = os.path.join(self.procfs_path, bus, "{}.{}".format(device, fn))
+        try:
+            with open(current_file, 'rb') as f:
+                f.seek(2)
+                data = f.read(2)
```

Comment:
@assrinivasan does this result in real pcie transaction?

Reply:
This does not -- this gets the intended device ID of the PCIe device by querying the procfs data file directly, thereby bypassing any SCD driver hijack or pcie*.yaml file weirdness. I added the actual transaction check in the latest commit.